syntax = "proto3";

package syft.core.node.common.service;

import "proto/core/common/common_object.proto";
import "proto/core/io/address.proto";

message ReprMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
}


message CreateVMMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  string settings = 3;
  syft.core.io.Address reply_to = 4;
}

message CreateVMResponseMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  syft.core.io.Address vm_address = 3;
  bool success = 4;
  string msg = 5;
}

message CreateWorkerMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  string settings = 3;
  syft.core.io.Address reply_to = 4;
}

message CreateWorkerResponseMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  syft.core.io.Address vm_address = 3;
  bool  success = 4;
  string msg = 5;
}
